storage/storage_interface.ml

```diff
@@ -257,8 +293,8 @@ module DP = struct
 	external destroy: dbg:debug_info -> dp:dp -> allow_leak:bool -> unit = ""
 
 
-	(** [attach_info task sr vdi dp]: returns the params of the dp (the return value of VDI.attach) *)
-	external attach_info: dbg:debug_info -> sr:sr -> vdi:vdi -> dp:dp -> attach_info = ""
+	(** [attach_info task sr vdi dp]: returns the params of the dp (the return value of VDI.attach2) *)
+	external attach_info: dbg:debug_info -> sr:sr -> vdi:vdi -> dp:dp -> backend = ""
```

Comment:
I think this would break compatibility with xapi, could you link to this PR all the other PRs required to make everything compile and work? (and mark the PRs as "don't merge" since they all need to be merged at once)